internal/datastore/postgres/stats.go

```diff
@@ -45,7 +45,7 @@ func (pgd *pgDatastore) Statistics(ctx context.Context) (datastore.Stats, error)
 	var uniqueID string
 	var nsDefs []datastore.RevisionedNamespace
 	var relCount int64
-	if err := pgd.dbpool.BeginTxFunc(ctx, pgd.readTxOptions, func(tx pgx.Tx) error {
+	if err := pgd.readPool.BeginTxFunc(ctx, pgd.readTxOptions, func(tx pgx.Tx) error {
```

Comment:
I feel like the reason conn pool where split are less about "read and write" and more about "fast TX / slow TX". If our transactions in CRDB were faster, we wouldn't even need to do this split in the first place.

For the sake of the argument, this transaction here - if I recall correctly - is rather slow, and could put the same kind of contention over the read-connection pool that write TX did.

Reply:
> If our transactions in CRDB were faster, we wouldn't even need to do this split in the first place.

I don't think that this is true. Regardless of how fast writes could be, they'll still always be in a different order of magnitude than read queries. There will always be some scale where you'll end up seeing the writes contend reads for connections. This PR acknowledges that reality.

The ultimate fix for this issue is to use a transport that allows for multiplexing queries over a single connection (e.g. gRPC). The problem is that many SQL databases have stateful connection sessions, so they cannot reuse the connection until they've gotten a response.